.github/workflows/publish.yml

```diff
@@ -1,65 +1,82 @@
-name: Publish Python 🐍 distributions 📦 to PyPI
+name: Build and publish Python 🐍 distributions 📦 to PyPI
 
 on:
   release:
-    types: [published]
+    types: [ published ]
   push:
     branches:
       - main
 
 jobs:
+  test:
+    name: Run Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.10'
+
+      - name: Install test dependencies
+        run: python -m pip install -r requirements/local.txt
+
+      # Run tests
+      - name: Run tests
+        run: pytest
+
```

Comment:
we have two options

First: make this job as a mandate before we build/pubilsh so when fails, so no other work should be done 

Second: make it optional, so I think after do not take time we can add as step to next job **so not checkout and install deps twice**